### PR TITLE
EFRS-731 Refactor NGRX code

### PR DESCRIPTION
**Pull request by endypanda**

### CODE REVIEW

File: ui/src/app/features/application-header/application-header.facade.ts [hunk 1]
```diff
@@ -16,17 +16,14 @@
...
-
+import { Observable, Subscription } from 'rxjs';
 import { Application } from '../../data/interfaces/application';
 import { IFacade } from '../../data/interfaces/IFacade';
 import { AppState } from '../../store';
 import { deleteApplication, updateApplication } from '../../store/application/action';
-import { selectCurrentApp, selectCurrentAppId, selectUserRollForSelectedApp } from '../../store/application/selectors';
+import { selectCurrentApp, selectCurrentAppId} from '../../store/application/selectors';
 import { selectIsLoadingApplicationList } from '../../store/user/selectors';
+import {selectUserRole} from '../../store/app-user/selectors';
```
Comment (isobolev) [inline ui/src/app/features/application-header/application-header.facade.ts:26]: please format

File: ui/src/app/store/app-user/selectors.ts [hunk 2]
```diff
@@ -13,12 +13,58 @@
  * or implied. See the License for the specific language governing
  * permissions and limitations under the License.
  */
-import { createFeatureSelector, createSelector } from '@ngrx/store';
+import {createFeatureSelector, createSelector} from '@ngrx/store';
```
Comment (isobolev) [inline ui/src/app/store/app-user/selectors.ts:16]: please format

File: ui/src/app/store/app-user/reducers.ts [hunk 3]
```diff
@@ -58,7 +57,7 @@ const reducer: ActionReducer<AppUserEntityState> = createReducer(
       { ...state, isPending: false }
     )
   ),
-  on(updateAppUserRoleFailAction, state => ({ ...state, isPending: false })),
+  on(updateAppUserRoleFail, state => ({ ...state, isPending: false })),
```
Comment (isobolev) [inline ui/src/app/store/app-user/reducers.ts:60]: can be combined with updateAppUserRoleFail handler above

File: ui/src/app/store/role/reducers.ts [hunk 4]
```diff
@@ -30,9 +30,9 @@ export const initialState: RoleEntityState = roleAdapter.getInitialState({
 
 const reducer: ActionReducer<RoleEntityState> = createReducer(
   initialState,
-  on(loadRolesEntityAction, state => ({ ...state, isPending: true })),
-  on(setPendingRoleEntityAction, (state, { isPending }) => ({ ...state, isPending })),
-  on(fetchRolesEntityAction, (state, { role }) => {
+  on(loadRolesEntity, state => ({ ...state, isPending: true })),
+  on(setPendingRoleEntity, (state, { isPending }) => ({ ...state, isPending })),
+  on(fetchRolesEntity, (state, { role }) => {
```
Comment (isobolev) [inline ui/src/app/store/role/reducers.ts:35]: > refactor fetchRolesEntityAction in reducer and similar reducer functions where we use removeAll and then addOne. setAll or setOne can be used

**Review by isobolev — COMMENTED** [review]
looks much better, just a few comments. Please also double check these AC:
> add missing error handling (catchError and display error in toast message if any)
reducers

> review reducers code to improve it using appropriate adapter functions

**Comment by isobolev** [comment]
@endypanda please address comments above

**Review by pbohdan — APPROVED** [review]

**Pull request merged**